### PR TITLE
Fix enthralling hold

### PR DESCRIPTION
**Pull request by htrajan**
Fixes issue with Enthralling Hold mentioned in https://github.com/magefree/mage/issues/6646

### CODE REVIEW

File: Mage.Sets/src/mage/cards/e/EnthrallingHold.java [hunk 1]
```diff
@@ -62,3 +54,16 @@ public EnthrallingHold copy() {
         return new EnthrallingHold(this);
     }
 }
+
+class EnthrallingHoldTarget extends TargetCreaturePermanent {
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/e/EnthrallingHold.java:58]: Missing override copy constructor. Original card (DreamLeash) must be fixed too.

Comment (htrajan) [reply Mage.Sets/src/mage/cards/e/EnthrallingHold.java:58]: fixed

**Comment by Zerrisx** [comment]
Also, this is still bugging if the creature is untapped with Enthralling Hold still on the stack.
image

**Comment by JayDi85** [comment]
Maybe it can be replaced as normal `TargetCreaturePermanent()` with untapped creature filter instead custom target.

**Comment by htrajan** [comment]
Hmm, very odd. I didn't test it again after making your suggested changes, @JayDi85 but looks like with your suggestion it still fizzles; without implementing copy constructor it does not fizzle but I can also choose an untapped creature as the target. (So I guess Dream Leash was not initially implemented correctly either.)

**Comment by htrajan** [comment]
I am trying a fix to change to a possibleTargets override rather than a canTarget override

**Comment by JayDi85** [comment]
Hmm, I've got the idea behind that strange code -- it's not target restriction, it's a static restriction (so target must checks only permanent legality, not tapped/untapped). [Rules forum about Dream Leash](https://www.mtgsalvation.com/forums/magic-fundamentals/magic-rulings/magic-rulings-archives/253345-dream-leash).

That link must be added to cards.

**Comment by htrajan** [comment]
Is there a way to only check selection legality and not target legality (to avoid fizzle)? I couldn't find an appropriate method on Target/TargetImpl/TargetPermanent to override.

**Comment by JayDi85** [comment]
override target's `boolean isLegal(Ability source, Game game);`

**Comment by htrajan** [comment]
Yeah, I think that works. Lots of dupe code but maybe fine for now. My proposal is add new method to Target: boolean isStillLegal. In TargetImpl, this defaults to canTarget. We use it instead of canTarget in TargetImpl::isLegal.

**Comment by JayDi85** [comment]
That's good solution. Can you add unit test for that too (normal cast, untapped target -> no fizzle, illegal target -> fiizzled)?

**Pull request merged**